homeassistant/components/sensor/steam_online.py

```diff
@@ -75,24 +78,38 @@ def update(self):
         """Update device state."""
         try:
             self._profile = self._steamod.user.profile(self._account)
-            if self._profile.current_game[2] is None:
-                self._game = 'None'
-            else:
-                self._game = self._profile.current_game[2]
+            self._game = self._get_current_game()
             self._state = {
                 1: STATE_ONLINE,
                 2: STATE_BUSY,
                 3: STATE_AWAY,
                 4: STATE_SNOOZE,
-                5: STATE_TRADE,
-                6: STATE_PLAY,
-            }.get(self._profile.status, 'Offline')
+                5: STATE_LOOKING_TO_TRADE,
+                6: STATE_LOOKING_TO_PLAY,
+            }.get(self._profile.status, STATE_OFFLINE)
             self._name = self._profile.persona
             self._avatar = self._profile.avatar_medium
         except self._steamod.api.HTTPTimeoutError as error:
             _LOGGER.warning(error)
             self._game = self._state = self._name = self._avatar = None
 
+    def _get_current_game(self):
+        game_id = self._profile.current_game[0]
+        game_extra_info = self._profile.current_game[2]
+
+        if game_extra_info:
+            return game_extra_info
+
+        if game_id:
+            # Initialize app list before usage to benefit from internal caching
+            app_list = self._steamod.apps.app_list()
```

Comment:
It looks like the upstream lib will handle caching if we keep around a reference to this app list. Let's declare this in the constructor and use it here. So in `__init__`: `self._app_list = steamod.apps.app_list()`, and then here we can just do `if game_id in self.app_list:`.

Reply:
You are correct yes, I will make the change. 

One thing that I though of further would be to have it in the `setup_platform` function. As I've not written code for the project before I might be wrong here, but if I initialize the app_list within there, and pass it to the `SteamSensor` within the `add_devices`, that would eliminate even further api calls and more only one app_list would be used right? Instead of initializing one for each steam user that has been configured, that is.

Reply:
Good call! 👍